src/internal_events/file.rs

```diff
@@ -82,61 +82,90 @@ mod source {
     use std::{io::Error, path::Path, time::Duration};
 
     use file_source::FileSourceInternalEvents;
-    use metrics::counter;
+    use metrics::{counter, register_counter, Counter, SharedString};
 
     use super::{FileOpen, InternalEvent};
     use crate::emit;
-    use vector_common::internal_event::{error_stage, error_type};
+    use vector_common::internal_event::{
+        error_stage, error_type, ByteSize, InternalEventHandle, RegisterInternalEvent,
+    };
 
     #[derive(Debug)]
-    pub struct FileBytesReceived<'a> {
-        pub byte_size: usize,
-        pub file: &'a str,
+    pub struct FileBytesReceived {
+        pub file: SharedString,
     }
```

Comment:
We'd like you to use the `registered_event` macro for this. It takes care of some of the boilerplate required to support a registered event. See the example in `lib/vector-common/src/internal_event/mod.rs`.

Reply:
Fixed.